### PR TITLE
a formula should be an instance of itself

### PR DESCRIPTION
**Pull request by yoni206**
The side condition **is_inst** fails when checking whether **f** is an instance of **f**.
The reason is that the same **markvar** was used both for **t** and **ti**, that happen to be identical in this case. 
Therefore, when **eqterm** does a **markvar** on n1 (which is **t**), it unmarks **t**, and the conditional returns **ff**.
The fix (courtesy of Andres) is to use a different **markvar**, namely **markvar2**.

### CODE REVIEW

**Comment by 4tXJ7f** [comment]
I've discussed this issue and the issue of comparing arbitrary expressions with Andy, and we came up with the following fix for the default case of `is_inst_t`: `(ifmarked t (do (markvar t) (let r (is_inst_t ti k) (do (markvar t) r)) (eqterm ti t))`. Let's discuss tomorrow.

**Comment by yoni206** [comment]
added this change to the pull request.

**Comment by HanielB** [comment]
There are trailing spaces, long lines and tabs in the changes. I'm not sure which guidelines we should use for LFSC, but we should certainly avoid these things.

**Comment by ajreynol** [comment]
Don't forget to remove the new files you added for testing, otherwise, it looks good to me.

File: proofs/signatures/th_quant.plf [hunk 1]
```diff
@@ -3,18 +3,28 @@
                 (! f formula
                   formula))))
 
-(program eqterm ((n1 term) (n2 term)) bool
+(program eqvar ((n1 term) (n2 term)) bool
```
Comment (ajreynol) [inline proofs/signatures/th_quant.plf]: This function can be removed right?

**Pull request merged**